src/explorer.api/Controllers/ExploreController.cs

```diff
@@ -82,21 +79,39 @@ public class ExploreController : ControllerBase
                 return NotFound($"Couldn't find exploration with id {explorationId}.");
             }
 
+            var exploreResult = new ExploreResult(explorationId, exploration);
+
             if (explorationStatus != ExplorationStatus.New && explorationStatus != ExplorationStatus.Processing)
             {
                 try
                 {
-                    await explorationRegistry.Remove(explorationId);
+                    // Await the completion task to force exceptions to the surface.
+                    await exploration.Completion;
                 }
                 catch (TaskCanceledException)
                 {
                     // Do nothing, just log the occurrence.
                     // A TaskCanceledException is expected when the client cancels an exploration.
                     logger.LogInformation($"Exploration {explorationId} was canceled.", null);
                 }
+                catch (Exception)
+                    when (!(exploration.Completion.Exception is null))
+                {
+                    // Log any other exceptions from the explorer and add them to the response object.
+                    logger.LogWarning($"Exceptions occurred in the exploration tasks for exploration {explorationId}.");
+
+                    foreach (var innerEx in exploration.Completion.Exception!.Flatten().InnerExceptions)
+                    {
+                        exploreResult.AddErrorMessage(innerEx.Message);
```

Comment:
It might be useful to also log the complete exception here.

Reply:
I was hesitant to do this because there might be a lot of inner exceptions. Maybe just log the first few?

Reply:
But you add all the exception messages in the result, so I don't really see the problem with having all the exceptions in the log files. And I think that it is important to have all of them in the log because some exceptions might signal a problem in our code and the exception info might be useful when trying to reproduce the problem.

Reply:
Yes you're right, no reason not to log everything. 

0f66a70